web/src/settings_org.js

```diff
@@ -520,6 +520,9 @@ export function discard_property_element_changes(elem, for_realm_default_setting
             // Because this widget has a radio button structure, it
             // needs custom reset code.
             $elem.find(`input[value='${CSS.escape(property_value)}']`).prop("checked", true);
+
+            // Hide the stream privacy warning banner
+            settings_components.hide_stream_privacy_warning_compose_banner();
```

Comment:
Would it be better to just make a separate switch case for "stream_privacy" so that this code is not called for other settings.